### ISSUE TITLE
Relay internet radio via the user's cloud instance

### ISSUE THREAD

**Issue by paulijar** [owncloud/music#1035]
At time being, the playback of non-HLS internet radio streams and podcasts works so that the web front-end streams the content directly from its original source. It might be beneficial to relay the internet radio feeds instead via the user's ownCloud or Nextcloud instance. Especially, this would help to solve the issue of streaming HTTP content on a cloud instance which uses HTTPS connection; at the moment this requires tweaking the page security settings on the browser.

The downside of this would be that it would cause additional load and network traffic on the server. It should be considered if this should be made a configurable feature and/or used only when streaming from a HTTP source.

_Originally requested by @jacotec in https://github.com/owncloud/music/issues/1025#issuecomment-1324909665_

**Comment by jacotec** [comment]
I think the network traffic for audio streams is not really high on the server side ... on the pro side it would also solve geoblock issue when traveling,

**Comment by JoshuaPettus** [comment]
Unless you geoblocked your own nextcloud server for security. XD

I would support some sort of toggle.  For single user or small family nextcloud instances, I don't think the added bandwidth would be too much of a great issue.  We are already streaming regular music from the server. Usually these highly compressed podcast files aren't too much, and certainly better then the flac files I have.  For my residential Internet, it is the server upload that is the limited resource (which I've already accepted by having a NC server), not so much download.

**Comment by JackRoublard** [comment]
Just encountered the HTTP/HTTPS mixed content issue on my small 3-user instance. I'm all in favor of this solution, provided a toggle in the admin menu which defaults to off, so as to avoid overloading traffic on big instances

**Comment by paulijar** [comment]
Music v2.1.0 was just released and it now relays the radio and podcast streams via the cloud server. In the end, I decided to make this the default behavior and the admins need to opt out if they wish to use the old system. Also, to use the old direct streaming, the admins now need to separately whitelist the allowed stream source domains (or all domains with wildcards).

**Comment by jacotec** [comment]
> Music v2.1.0 was just released and it now relays the radio and podcast streams via the cloud server. In the end, I decided to make this the default behavior and the admins need to opt out if they wish to use the old system. Also, to use the old direct streaming, the admins now need to separately whitelist the allowed stream source domains (or all domains with wildcards).

This is awesome news! I'll check this out right tomorrow!

**Comment by jacotec** [comment]
@paulijar Can it be that the subsonic API isn't proxied? My network monitoring shows correct proxying when using the Nextcloud Webserver, but using the Play:Sub app via Nextcloud the stream runs toward the iDevice, not to the cloud server

**Comment by paulijar** [comment]
Yes, I didn't apply the relaying to the Subsonic or Ampache APIs at this time. I considered it but it would have taken non-negligible amount of extra effort on top of the already rather difficult and complex task. And I figured that the main reasons behind this change were to prevent the problems caused by mixed HTTP/HTTPS for browsers as well as the need to create "holes" to the Content Security Policy. Neither of these are issues for the Subsonic/Ampache clients.

Would you see it as beneficial to apply this to the Subsonic API? I may consider it in later versions, once we have some experience from the new system and know if it causes any new problems. The geoblocked streams while traveling could of course be an issue, although I see this as a bit niche case.

**Comment by jacotec** [comment]
Dear @paulijar ,

for me I'm using it 95% at iPhone/iPad where the NC web interface is not optimal ... I'm using the great Play:Sub app, so Subsonic API. Therefore I'd be more than happy if that can be proxied as well (I was hoping so long for the proxy feature and have been so happy reading your announcement, but for now it doesn't help me)
